language-design.md

```diff
@@ -49,6 +148,41 @@ In what way is an API a language?
 
 **Response**
 
+In _How to Design a Good API and Why it Matters_, Bloch lists maxims for 
+designing a good API. Many of these maxims also apply to languages. In one
+of these maxims, he states "If you get an API right, code will read like
+prose" [Bloch, 2006]. So similar to a language, an API should have fluency.
+
+Bloch also mentions that "When in doubt, leave it out" [Bloch, 2006]. This
+is similar to Steele's claim that a growable language should not have too
+many features. While Bloch states that "You have one chance to get [a public
+API] right", he also says that "You can always add things later" [Bloch, 2006].
+In this sense, an API — like a language — can grow. In fact, an API can be seen
+as an extension of a growing language, so it inherits some properties of the
+platform it is built on. Due to the same maxim, "When in doubt, leave it out[,]"
+an API shares commonalities with a "small language" as Steele defines it. It 
+has a small set of vocabulary which programmers can apply with some 
+degree of fluency.
+
+Bloch lists several maxims that relate to creating an intuitive API.
+"Names matter" and "the principle of least astonishment".
+This goes hand in hand with Pavlus's argument for languages that "were 
+better-designed" [Pavlus, 2012].
+However, this may also be an example of how an API is _not_ like a language.
+Pavlus mentions multiple common programming languages — such as Perl and Java —
+are not intuitive, as "The broad computer science academic community has not 
+paid a tremendous amount of attention to programming language usability"
+[Pavlus, 2012].
+It seems like an API strives to be intuitive more than a language does.
+This is probably an easier goal for an API to aim for since an API designer
+can assume a level of competence in whatever language the API uses in the 
+users of the API.
+
+Another way that an API is not like a language is that an API is expected to be
+self-documenting, "it should rarely require documentation to read code written
+to a good API" [Bloch, 2006]. While a programmer who has never used an API
+before should be able to understand code written to it, the same may not be
+true for a programmer reading a language that is completely new to them.
```

Comment:
I would say that a language should also be fairly self documenting. There can be some exceptions, but I think the documentation should be mostly for figuring out how to do something you want, not figuring out something you saw.